packages/cashier/src/hooks/useWS.ts

```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react';
+import { useWS as useWSShared } from '@deriv/shared';
+import { TSocketCallTypes } from 'Types';
+
+type TSocketCalls = keyof TSocketCallTypes;
+
+type TResponse<T extends TSocketCalls> = TSocketCallTypes[T]['response'][T];
+
+type TRequestPropsExceptions = 'verify_email';
+
+type TRequestProps<T extends TSocketCalls> = Omit<
+    TSocketCallTypes[T]['request'],
+    (T extends TRequestPropsExceptions ? never : T) | 'passthrough' | 'req_id'
+>;
+
+type TRequest<T extends TSocketCalls> = TRequestProps<T> extends Record<string, never>
+    ? Record<string, never>
+    : TRequestProps<T>;
+
+const useWS = <T extends TSocketCalls>(name: T) => {
+    const [isLoading, setIsLoading] = useState(false);
+    const [error, setError] = useState<unknown>();
+    const [data, setData] = useState<TResponse<T>>();
+    const WS = useWSShared();
+
+    const send = async (props: TRequest<T>) => {
+        if (isLoading) return;
+
+        setIsLoading(true);
+
+        try {
+            const response = await WS.send({ [name]: 1, ...props });
+
+            if (response.error) {
+                setError(response.error);
+            } else {
+                setData(response[name]);
+            }
+        } catch (e) {
+            setError(e);
+        } finally {
+            setIsLoading(false);
+        }
+    };
+
+    return { send, isLoading, error, data };
+};
+
+export default useWS;
```

Comment:
This hook should be part of `@deriv/api` package and accessible for everyone who uses this package.

Reply:
as far as i know deriv-api doesn't have any react in it. and it mostly manage by BE team. so adding it to that package doesn't make sense to me.
are you sure that you mentioned the correct project?

Reply:
@ali-hosseini-fs I was not aware of the `@deriv/api` implementation, So if there is no React involved then it's better to keep it here as it is.

Optimistically, In the future we can have a `@deriv/hooks` package in our monorepo which can be shared between other packages. But as we discussed we have decided to keep everything within the cashier's scope for this stage.